phrosty/imagesubtraction.py

```diff
@@ -532,54 +567,77 @@ def swarp_coadd_img(imgpath_list,refpath,out_name,out_path=output_files_rootdir,
         else:
             imgpaths.append(p)
 
-    coadd_savedir = os.path.join(out_path,'coadd')
+    if psf:
+        subdir = 'coadd_psf'
+    elif not psf:
+        subdir = 'coadd'
+    coadd_savedir = os.path.join(out_path,subdir)
     check_and_mkdir(coadd_savedir)
     coadd_savepath = os.path.join(coadd_savedir,out_name)
+
+    print('COADD_SAVEPATH in swarp_coadd_img')
+    print(coadd_savepath)
+    print('FITS_obj')
+    print(imgpaths)
+    print('FITS_ref')
+    print(refpath)
+    print('OUT_path')
+    print(coadd_savepath)
     
     coadd = PY_SWarp.Coadd(FITS_obj=imgpaths, FITS_ref=refpath, ConfigDict=cd,
                             OUT_path=coadd_savepath, FILL_VALUE=np.nan)
 
-    return coadd_savepath, imgpaths
-
-def swarp_coadd_psf(ra,dec,sci_skysub_paths,sci_imalign_paths,
-                    ref_table,refpath,out_name,out_path=output_files_rootdir):
-    """_summary_
-
-    :param ra: _description_
-    :type ra: _type_
-    :param dec: _description_
-    :type dec: _type_
-    :param sci_imalign: _description_
-    :type sci_imalign: _type_
-    :param sci_skysub: _description_
-    :type sci_skysub: _type_
-    :param ref_table: filter, pointing, sca astropy table w/ images in the coadded template. 
-    :type ref_table: _type_
-    :param refpath: Path to coadded template. For WCS info. 
-    :type refpath: _type_
-    :param out_name: _description_
-    :type out_name: _type_
-    :param out_path: _description_, defaults to output_files_rootdir
-    :type out_path: _type_, optional
-    :return: _description_
-    :rtype: _type_
-    """
-
-    coadd_psf_savedir = os.path.join(out_path,'coadd_psf')
-    check_and_mkdir(coadd_psf_savedir)
-    coadd_psf_savepath = os.path.join(coadd_psf_savedir,out_name)
+    print('COADD_SAVEPATH IN swarp_coadd_img again')
+    print(coadd_savepath)
 
-    psf_list = []
-    # Can be parallelized: 
-    for i, row in enumerate(ref_table):
-        psf = rotate_psf(ra,dec,sci_skysub_paths[i],sci_imalign_paths[i],
-                         sci_band=row['filter'],sci_pointing=row['pointing'],sci_sca=row['sca'],
-                         ref_path=refpath)
-        psf_list.append(psf)
-
-    coadd_psf_init = swarp_coadd_img(psf_list,refpath,coadd_psf_savepath)
+    return coadd_savepath, imgpaths
 
-    return coadd_psf_savepath
+# def swarp_coadd_psf(ra,dec,sci_skysub_paths,sci_imalign_paths,
```

Comment:
Re-name this as a function you don't use, but I suggest moving away from the habit of commenting-out code that you might want later.  Instead define it all in a function name that describes what you might want to do with it in the future and then you can choose to use it or not.  In this case it can be as silly as `def old_swarp_coadd_psf` but you can probably do better with something like `def swarp_coadd_psf_with_weights` or something, I don't know.  You'll have better ideas then me.

Reply:
In this case, I actually think I don't want swarp_coadd_psf(). I think it's entirely unnecessary. You can change the weight stuff in swarp_coadd() with kwargs. I think it should just be one function that coadds whatever you input.